packages/esm-patient-list-app/src/offline-patient-table/offline-patient-table.component.tsx

```diff
@@ -43,70 +49,120 @@ export interface OfflinePatientTableProps {
+      age: patient.birthDate ? age(patient.birthDate) : '',
+    });
+
+    for (const patient of offlineRegisteredPatientsSwr.data ?? []) {
+      result.push(mapPatientToRow(patient, true));
+    }
+
+    for (const patient of offlinePatientsSwr.data ?? []) {
+      result.push(mapPatientToRow(patient, false));
+    }
+
+    return result;
+  }, [offlinePatientsSwr.data, offlineRegisteredPatientsSwr.data, store.offlinePatientDataSyncState]);
+
+  const handleUpdateSelectedPatientsClick = async (selectedRows: ReadonlyArray<DenormalizedRow>) => {
+    const offlinePatientUuidsToSync = selectedRows
+      .map((row) => row.id)
+      .filter((id) => offlinePatientsSwr.data.some((offlinePatient) => offlinePatient.id === id));
+    return await Promise.all(offlinePatientUuidsToSync.map((patientUuid) => syncOfflinePatientData(patientUuid)));
   };
 
-  const handleRemovePatientsFromOfflineListClick = async (selectedRows: Array<{ id: string }>) => {
+  const handleRemovePatientsFromOfflineListClick = async (selectedRows: ReadonlyArray<DenormalizedRow>) => {
     const closeModal = showModal('offline-tools-confirmation-modal', {
       title: t('offlinePatientsTableDeleteConfirmationModalTitle', 'Remove offline patients'),
       children: t(
         'offlinePatientsTableDeleteConfirmationModalContent',
-        'Are you sure that you want to remove all selected patients from the offline list? The charts will no longer be available in offline mode.',
+        'Are you sure that you want to remove all selected patients from the offline list? Their charts will no longer be available in offline mode and any newly registered patient will be permanently deleted.',
       ),
       confirmText: t('offlinePatientsTableDeleteConfirmationModalConfirm', 'Remove patients'),
       cancelText: t('offlinePatientsTableDeleteConfirmationModalCancel', 'Cancel'),
       closeModal: () => closeModal(),
       onConfirm: async () => {
-        const patientUuids = selectedRows.map((row) => row.id);
-        for (const patientUuid of patientUuids) {
-          await removePatientFromLocalPatientList(userId, offlinePatientListId, patientUuid);
-        }
+        const offlineRegisteredPatients = await getFullSynchronizationItems<{ fhirPatient: fhir.Patient }>(
+          'patient-registration',
+        );
+        const allPatientUuidsToBeDeleted = selectedRows.map((row) => row.id);
+        const offlinePatientUuidsToBeDeleted = allPatientUuidsToBeDeleted.filter((id) =>
+          offlinePatientsSwr.data.some((patient) => patient.id === id),
+        );
+        const offlineRegisteredPatientUuidsToBeDeleted = allPatientUuidsToBeDeleted.filter((id) =>
+          offlineRegisteredPatientsSwr.data.some((patient) => patient.id === id),
+        );
+
+        await Promise.all(
+          offlinePatientUuidsToBeDeleted.map((patientUuid) =>
+            removePatientFromLocalPatientList(userId, offlinePatientListId, patientUuid),
+          ),
+        );
```

Comment:
I'm not understanding why these need to be async functions? Isn't this just removing patient uuids locally, and when we call it we `await`, so it becomes synchronous again?

Reply:
That's more or less the case, but the UUIDs are stored in the IndexedDB whose API is asynchronous (`removePatientFromLocalPatientList` is therefore `async` as well).